Bio/PDB/__init__.py

```diff
@@ -88,3 +97,67 @@
     from .SASA import ShrakeRupley
 except ImportError:
     pass
+
+_FormatToParser = {"pdb": PDBParser(), "cif": MMCIFParser()}
+
+
+def read(handle, format, id=None):
+    """Parse a file or handle using one of the available parsers to generate a structure.
+
+    Arguments:
+     - handle    - handle to the file, or the filename as a string
+     - format    - string describing the file format. can be [pdb, cif, mmtf]
+     - id        - optional id of the structure, if None header information or then base name is used
+
+    If you have the file name in a string 'filename', use:
+
+    >>> from Bio import PDB
+    >>> filename = "1A8O.pdb"
+    >>> PDB.read(filename, format="pdb")
+
+    The function also supports file handles and gzip files
+
+    >>> from Bio import PDB
+    >>> PDB.read("1A8O.pdb.gz", format="pdb")
+    """
+    # TODO *kwargs for the parser to allow QUIET or permissive mode
+
+    # Try and give helpful error messages:
+    if not isinstance(format, str):
+        raise TypeError("Need a string for the file format (lower case)")
+    if not format:
+        raise ValueError("Format required (lower case string)")
+    if format != format.lower():
+        raise ValueError("Format string '%s' should be lower case" % format)
+
+    # need to use different parser for mmtf versus pdb,cif
+    if format == "mmtf":
+        # TODO: Raise warning if ID is set. MMTFParser does not support passing an id
+        # MMTF needs a seperate parser as get_structure does only take 1 positional argument and does not support file handles
+        if isinstance(handle, str):
+            parser = MMTFParser()
+            return parser.get_structure(handle)
+        else:
+            raise TypeError("'%s' parser only accepts str not handle" % format)
+
+    elif format in _FormatToParser:  # Map the file format to a parser
+        # if we are provided a path to a file, check if gzipped,
+        # else open handle directly
+        if isinstance(handle, str):
+            encoding = guess_type(handle)[1]  # uses file extension
+            _open = partial(gzip.open, mode="rt") if encoding == "gzip" else as_handle
+        else:
+            _open = as_handle
```

Comment:
Do we still need `as_handle` here? If `open` is sufficient, then I would prefer that is it makes it easier to follow the logic in this code section.

Reply:
`guess_type` will fail if a handle is passed. I will change it together with the the `try: else:` block to make the logical flow better.

Reply:
I like the approach with `guess_type`. You could consider what to do if `gzip.open` fails because the file turned out not to be gzipped after all. But maybe this possibility is too remote and we can just ignore it.

Reply:
see above. I will add a check for bad gzip files once we have decided on a way to to check if it is a gzip file ( via magic, guess_type or else).